#! /bin/sh
patch -p1 -f -l $* < $0
exit $?

https://codereview.qt-project.org/c/qt/qtbase/+/695395

From 583c383d7111e472a972e7d1bcb392922744bdf9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ingo=20Kl=C3=B6cker?= <dev@ingo-kloecker.de>
Date: Tue, 2 Dec 2025 13:57:32 +0100
Subject: [PATCH] a11y uia: Notify about state changes of checkable item view
 items

To allow assistive technology on Windows to react to changes of the
checked state of item view items send appropriate notifications also
for the Cell, ListItem and TreeItem roles.

Fixes: QTBUG-141856
Change-Id: I5ff341fa260ea95c0cf80bc25c141f9da0ea54da
Reviewed-by: Michael Weghorn <m.weghorn@posteo.de>
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
---
 .../windows/uiautomation/qwindowsuiamainprovider.cpp          | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp b/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
index fc0e053f396..4fd9f8c6293 100644
--- a/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
+++ b/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
@@ -88,8 +88,8 @@ void QWindowsUiaMainProvider::notifyStateChange(QAccessibleStateChangeEvent *eve
 {
     if (QAccessibleInterface *accessible = event->accessibleInterface()) {
         if (event->changedStates().checked || event->changedStates().checkStateMixed) {
-           // Notifies states changes in checkboxes.
-           if (accessible->role() == QAccessible::CheckBox) {
+            // Notifies states changes in checkboxes and checkable item view items.
+            if (accessible->role() == QAccessible::CheckBox || accessible->role() == QAccessible::Cell || accessible->role() == QAccessible::ListItem || accessible->role() == QAccessible::TreeItem) {
                 if (auto provider = providerForAccessible(accessible)) {
                     long toggleState = ToggleState_Off;
                     if (accessible->state().checked)
--
2.52.0
